domino/airflow/_operator.py

```diff
@@ -0,0 +1,165 @@
+    python client.
+
+
+    Notes
+    ------
+    When combining `isDirect=True` with a command, you
+    need to pass in the entire command as a single string
+    in the command array. 
+    """
+
+    template_fields = ("command", "title")
+    ui_color = "#5188c7"
+
+    @apply_defaults
+    def __init__(
+        self,
+        project: str,
+        command: List[str],
+        host: Optional[str] = None,
+        api_key: Optional[str] = None,
+        domino_token_file: Optional[str] = None, 
+        isDirect: bool = None,
+        commitId: Optional[str] = None,
+        title: Optional[str] = None,
+        tier: Optional[str] = None,
+        publishApiEndpoint: Optional[bool] = None,
+        poll_freq: Optional[int] = 5,
+        max_poll_time: Optional[int] = 6000,
+        startup_delay: Optional[int] = 10,
+        include_setup_log: Optional[bool] = True,
+        *args,
+        **kwargs
+    ):
+        super(DominoOperator, self).__init__(*args, **kwargs)
+
+        self.log.info("Initializing Client...")
+        
+        self.project = project
+        self._api_key = api_key
+        self._host = host
+        self._domino_token_file = domino_token_file
+        self.command = command
+        self.is_direct = isDirect
+        self.commmit_id = commitId
+        self.title = title
+        self.tier = tier
+        self.publish_api_endpoint = publishApiEndpoint
+        self.poll_freq = poll_freq
+        self.max_poll_time = max_poll_time
+        self.startup_delay = startup_delay
+        self.include_setup_log = include_setup_log
+
+        self.client = None
+        self.run_id = None
+
+    def get_run_log(self) -> str:
+
+        url = self.client._routes.runs_stdout(self.run_id)
+
+        logs = list()
+
+        if self.include_setup_log:
+            logs.append(self.client._get(url)["setup"])
+
+        logs.append(self.client._get(url)["stdout"])
+
+        return "\n".join(logs)
+
+    def execute(self, context: Any) -> dict:
+
+        self.client = Domino(self.project, self._api_key, self._host, self._domino_token_file)
+        self.log.info("Client Initialized for project: %s", self.project)
+
+        self.log.info("Starting run.")
+        if self.startup_delay:
+            time.sleep(self.startup_delay)
+
+        if self.is_direct:
+            if len(self.command) > 1:
+                raise ValueError(
+                    "Domino API will not accept a "
+                    "multipart command string this long if is_direct=True"
+                )
+
+        run_response = self.client.runs_start(
+            command=self.command,
+            isDirect=self.is_direct,
+            commitId=self.commmit_id,
+            title=self.title,
+            tier=self.tier,
+            publishApiEndpoint=self.publish_api_endpoint,
+        )
+
+        self.run_id = run_response["runId"]
+
+        poll_start = time.time()
+        while True:
```

Comment:
Hey, Can we use `run_start_blocking` method instead of using `runs_start`, I see the only difference is the logs which we are passing into airflow. Can we change it something like below

```
run_response = self.client.runs_start_blocking(<arguments-here>)
log = self.get_run_log()
# spool out and replay entire log
      for line in log.splitlines():
            if line:
            # using bs4 to strip the HTML tags
                  text = BeautifulSoup(line).text
                  if "text-danger" in line:
                       self.log.warning(text)
                  else:
                        self.log.info(text)
return run_response
```

Reply:
Ah yes, since logs are only made available after a run completes and we can't pull them async this makes sense. I think originally I had planned to do that but realized it wasn't possible. I'll change it.